src/module.ts

```diff
@@ -3418,16 +3418,27 @@ export class SwitchBuilder {
     this.condition = condition;
   }
 
+  /** Links a case to the specified branch, replace old case if it is linked. */
+  addOrReplaceCase(value: i32, code: ExpressionRef[]): void {
+    let valueIndex = this.values.indexOf(value);
+    if (valueIndex >= 0) this.indexes[valueIndex] = this.addCode(code);
+    else this.addCase(value, code);
+  }
+
   /** Links a case to the specified branch. */
   addCase(value: i32, code: ExpressionRef[]): void {
```

Comment:
`addCase` quite small and uses only once. I guess better to remove it and use their body's code in `addOrReplaceCase` directly.